addons/sale_service/models/sale_order_line.py

```diff
@@ -94,11 +94,14 @@ def _additional_name_per_id(self):
 
         return name_per_id
 
-    def _name_search(self, name, domain=None, operator='ilike', limit=None, order=None):
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        domain = args or []
         # optimization for a SOL services name_search, to avoid joining on sale_order with too many lines
         if domain and ('is_service', '=', True) in domain and operator in ('like', 'ilike') and limit is not None:
-            query = self.env['sale.order.line']._search(domain, limit=limit, order=None)
+            query = self._search(domain, limit=limit)
             query.order = f'{query.table}.order_id DESC, {query.table}.sequence, {query.table}.id'
-            return query
-        else:
-            return super()._name_search(name, domain, operator, limit, order)
+            fields_to_fetch = self._determine_fields_to_fetch(['display_name'])
+            sols = self._fetch_query(query, fields_to_fetch)
```

Comment:
If we could force ordering by order _id_ instead of order _name_, then this could be simplified as
```py
sols = self.search_fetch(domain, ['display_name'], limit=limit, order='order_id.id DESC, sequence, id')
```

Reply:
Ordering on related fields is not yet implemented, so I kept the logic as is.

Reply:
I just added a commit for this.